stanford_soe_r25.module

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @file
+ * Code for the Stanford SOE R25 module.
+ */
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function stanford_soe_r25_form_stanford_r25_reservation_alter(&$form, &$form_state, $form_id) {
+  // We need to alter the reservation form and add field settings.
+  // The settings for "Event Attributes" 191,144,178,198,141, and 166.
+  $form['stanford_r25_booking_tentative']['#weight'] = -11;
+  $form['stanford_r25_booking_date']['#weight'] = -10;
+  $form['stanford_r25_booking_duration']['#weight'] = -9;
+  $form['stanford_r25_booking_enddate']['#weight'] = -8;
+  $form['stanford_r25_booking_reason']['#weight'] = -7;
+  $form['stanford_r25_booking_attr191']['#weight'] = -3;
+  $form['stanford_r25_booking_headcount']['#weight'] = -2;
+  $form['stanford_r25_booking_attr144']['#weight'] = -1;
+  $form['stanford_r25_booking_reason']['#title'] = t('Event Title');
+  $form['stanford_r25_booking_attr182']['#title'] = t('Event Details');
+  $form['stanford_r25_booking_attr191']['#title'] = t('Primary Organization');
+  $form['stanford_r25_booking_attr141']['#type'] = 'radios';
+  $form['stanford_r25_booking_attr141']['#options'] = array(1 => t('Yes'), 0 => t('No'));
+  $form['stanford_r25_booking_attr141']['#default_value'] = variable_get('stanford_r25_booking_attr141', 0);
+  $form['stanford_r25_booking_attr166']['#type'] = 'radios';
+  $form['stanford_r25_booking_attr166']['#options'] = array(1 => t('Yes'), 0 => t('No'));
+  $form['stanford_r25_booking_attr166']['#default_value'] = variable_get('stanford_r25_booking_attr141', 0);
+  $form['stanford_r25_booking_attr178']['#suffix'] = '<div class="form-item"><span class="description">' . t('If unsure, please put N/A') . '</span></div>';
+  $form['submit']['#prefix'] = '<div class="form-item"><span class="readandagree">' . t('By clicking the Request button, I confirm that I have read and agree to the') . ' ' . l(t("Booking Agreement"), 'https://insidesoe.stanford.edu/science-and-engineering-quad/policies-rates') . '.</span></div>';
+  $form['stanford_r25_booking_attr144']['#suffix'] = '<div class="form-item"><span class="description">' . t('E.g., 1234567-123-ABCDE (Case sensitive)') . '</span></div>';
+  $form['stanford_r25_booking_attr198']['#suffix'] = '<div class="form-item"><span class="description">' . t('E.g. Reservation spans multiple days; Breakout spaces needed') . '</span></div>';
+  $form['stanford_r25_booking_date']['#suffix'] = '<div class="starttime">' . t('Start time includes set-up time') . '</div>';
+  $form['stanford_r25_booking_enddate']['#suffix'] = '<div class="endtime">' . t('End time includes break-down time') . '</div>';
+  $form['stanford_r25_booking_attr182']['#maxlength'] = 80;
+
+  // We need this to be first for the verification to happen
+  // before the room booking takes place.
```

Comment:
Does the booking happen during verify? My former question is why does this have to be first in the verification functions? You're using array_unshift to put it first and it wasn't clear to what impact the other verification functions had on this.